### ISSUE TITLE
Problem: pg_yregress not collecting notices done during commit (deferred triggers)

### ISSUE THREAD

**Issue by yrashk** [omnigres/omnigres#261]
Given a deferred trigger that may issue a notice, when we have a `commit: true` test, these notices are seemingly not collected by pg_yregress.

This impedes testing of the trigger.

Proposed solution: ensure we collect notices issued at commit time

---

**Bounty size: $100**

[Read more about bounties](https://github.com/omnigres/omnigres/wiki/Bounties)

**Comment by rmodpur** [comment]
I tested with this
```yaml
$schema: "https://raw.githubusercontent.com/omnigres/omnigres/master/pg_yregress/schema.json"

instance:
  init:
  - create table test(id int)
  - |
    create function raise_notice() returns trigger as $$
    begin
      raise notice 'raising notice';
      return new;
    end;
    $$ language plpgsql
  - |
    create constraint trigger test_trigger after insert on test
    deferrable initially deferred
    for each row execute function raise_notice()
  
tests:
- name: notices
  query: insert into test values (1)
  commit: true
  notices:
  - raising notice
```
The test is **passing** which implies notices are being collected. Is there an example where it is not able to collect notices ?

Another thing i observed is the test still passes when removing `commit: true` line. There is no notice raised on aborting a tx, are expected notices checked only when actual output contains notices ?

**Comment by rmodpur** [comment]
@yrashk is this still a bug ? From what i have tested doesn't seem like it.

**Comment by yrashk** [comment]
What was your test? I don't recall fixing it yet, but I'm curious what was your test to see if the issue is defined accurately.

**Comment by rmodpur** [comment]
> What was your test? I don't recall fixing it yet, but I'm curious what was your test to see if the issue is defined accurately.

https://github.com/omnigres/omnigres/issues/261#issuecomment-1809943345

**Comment by yrashk** [comment]
Oh. I apologize, I was looking at the notification alone and not the whole issue. My bad. I'll get back to you shortly.

**Comment by yrashk** [comment]
The best context I can provide at the mlment is this:

To the best of my recollection, I ran into this issue when I was doing a multistep test (`steps`)

But the exact shape is escaping me at the moment.

**Comment by rmodpur** [comment]
@yrashk multistep tests always pass even if notices don't match, could you please assign it to me

**Comment by yrashk** [comment]
Solved in #383. 🎊 🍸
